build/Dockerfile

```diff
@@ -174,24 +224,30 @@ apt-get install -y --no-install-recommends \
 rm -rf /var/lib/apt/lists/*
 EOF
 
-# Copy Rust binaries
-ARG BASE_PATH
-WORKDIR ${BASE_PATH}/bin
+# Copy Rust binaries.
+# Explicitly copy each binary to avoid adding unnecessary files to the runtime image.
 ARG RUST_BUILD_PATH
-COPY --from=rust-builder ${RUST_BUILD_PATH}/target/release/cartesi-rollups-* .
-ENV PATH="${BASE_PATH}/bin:${PATH}"
-
-# Copy Go binary
-ARG GO_BIN_PATH
-COPY --from=go-builder ${GO_BIN_PATH}/cartesi-rollups-node .
+ARG RUST_TARGET=${RUST_BUILD_PATH}/target/release
+COPY --from=rust-builder ${RUST_TARGET}/cartesi-rollups-advance-runner /usr/bin
+COPY --from=rust-builder ${RUST_TARGET}/cartesi-rollups-authority-claimer /usr/bin
+COPY --from=rust-builder ${RUST_TARGET}/cartesi-rollups-dispatcher /usr/bin
+COPY --from=rust-builder ${RUST_TARGET}/cartesi-rollups-graphql-server /usr/bin
+COPY --from=rust-builder ${RUST_TARGET}/cartesi-rollups-host-runner /usr/bin
+COPY --from=rust-builder ${RUST_TARGET}/cartesi-rollups-indexer /usr/bin
+COPY --from=rust-builder ${RUST_TARGET}/cartesi-rollups-inspect-server /usr/bin
+COPY --from=rust-builder ${RUST_TARGET}/cartesi-rollups-state-server /usr/bin
+
+# Copy Go binary.
+ARG GO_BUILD_PATH
+COPY --from=go-builder ${GO_BUILD_PATH}/cartesi-rollups-node /usr/bin
 
-# Setup runtime dir
-ARG RUNTIME_DIR
-RUN <<EOF
-mkdir -p ${RUNTIME_DIR}
-chown cartesi:cartesi ${RUNTIME_DIR}
-EOF
+# Setup runtime dir.
+ARG RUNTIME_DIR=/usr/share/rollups-node
 WORKDIR ${RUNTIME_DIR}
+RUN chown cartesi:cartesi ${RUNTIME_DIR}
 
+# Set user back to low-privilege.
 USER cartesi
-CMD ["cartesi-rollups-node"]
+
+# Set the Go supervisor as the entrypoint.
```

Comment:
I suggest not using the naming entrypoint here to not confuse with Dockerfile's `ENTRYPOINT` instrunction.